dex/testing/dcr/harness.sh

```diff
@@ -89,19 +89,53 @@ chmod +x "${NODES_ROOT}/harness-ctl/mine-beta"
 # Reorg script
 cat > "${NODES_ROOT}/harness-ctl/reorg" <<EOF
 #!/bin/sh
+REORG_NODE="alpha"
+VALID_NODE="beta"
+REORG_DEPTH=1
+
+if [ "\$1" = "beta" ]; then
+  REORG_NODE="beta"
+  VALID_NODE="alpha"
+  REORG_DEPTH=3
+fi
+
+if [ "\$2" != "" ]; then
+  REORG_DEPTH=\$2
+fi
+
+# TODO: Cannot currently cause a 1+ block re-org on alpha because
+# beta cannot mine more than 1 main chain block while disconnected
+# from alpha.
+if [ "\${REORG_NODE}" = "alpha" ] && [ \${REORG_DEPTH} -gt 1 ]; then
+  echo "Cannot cause a re-org of more than 1 block on alpha."
+  exit 1
+fi
```

Comment:
This can be fixed following @davecgh's recommendation [here](https://github.com/decred/dcrdex/issues/646#issuecomment-683225733) to use the same seed and enable voting on both alpha and beta wallets, to ensure beta wallet can vote on tickets purchased with the alpha wallet after the nodes are disconnected.

I'm hesitant to do that because it could cause issues for simnet tests relying on these wallets having different different utxo sets and balances.

Reply:
Would need to create a new wallet instead of replacing w-beta.

But I'm not sure that we're really creating a true reorg with the current change.  I think we have to hook up a voting wallet to the beta node. @davecgh is it not actually a reorg with only a single block on the beta node (where there's no voting wallet).  @itswisdomagain says he's creating a reorg by mining a single block on the beta node with the votes received prior to disconnecting alpha and beta, and then causing a reorg on reconnect by mining more blocks.

Reply:
You can create a single block reorg without hooking up a wallet to the second node if you mine one block on the second node, and two on the first, but it may or may not work for the desired goal.

e.g.

* Starting state:
α: tip A (height: 100), mempool has votes for A
β: tip A (height: 100), mempool has votes for A
* Disconnect them and mine a block on β using the votes it has in the mempool
* β state is now: A (100) -> B (101), mempool does NOT have votes for B
* Mine a block on α
* α state is now: A (100) -> B' (101), mempool has votes for B'
* Mine another block on α
* α state is now: A (100) -> B' (101) -> C (102), mempool has votes for C
* Reconnect the nodes
* β notices α has more work and reorgs to state: A (100) -> B' (101) -> C (102)
* β also now gets the votes for C from α's mempool

However, notice this means the reorg happen on β, not α, which may or may not be acceptable.  Typically, testing works from the viewpoint of α, because it has the wallet connected, and there is no way to force the reorg to happen on **α** without doing what I described where more blocks are mined on β than α via having a second wallet that can independently deliver the votes.

Reply:
Here's how I've got reorg to happen on α:
Starting state:
- α: tip A (height: 100), mempool has votes for A
- β: tip A (height: 100), mempool has votes for A
- Disconnect them and mine a block on β using the votes it has in the mempool
- β state is now: A (100) -> B (101), mempool does NOT have votes for B
- Mine a block on α
- α state is now: A (100) -> B' (101), mempool has votes for B'
- Reconnect the nodes
- β now has votes for B
- Mine another block on β
- β state is now: A (100) -> B (101) -> C (102), mempool has votes for C
- α notices β has more work, purges B' (101) and reorgs to state: A (100) -> B (101) -> C (102)

Does that look correct? It works that way on testing though.

Reply:
That will work as long as both blocks have all 5 votes since it reverts to first-seen behavior in that case, however, it won't work in the general case when you start testing with less than 5 votes on some blocks.  As long as that is acceptable for this particular case of the DEX tests, then that should be fine.

Reply:
Hmm, I see.
- Assuming B (101) on node β has 3 votes, and B' (101) on node α has 5 votes.
- β mines another block C (102), so β is now A (100) -> B (101) -> C (102)
- Could β end up purging B (101) because it has fewer votes and become A (100) -> B' (101) -> C (102)?

Reply:
Yep, that's what will happen.  When you connect the nodes back up, β will learn about B' from α and since B' has more votes, the miner on β will choose to mine on B' instead resulting in A (100) -> B' (101) -> C (102) which will force a reorg on β, but not α.